packages/slate/src/interfaces/element.ts

```diff
@@ -18,9 +18,24 @@ export interface ElementInterface {
   isElement: (value: any) => value is Element
   isElementList: (value: any) => value is Element[]
   isElementProps: (props: any) => props is Partial<Element>
+  isElementType: (
+    value: any,
+    type: string
+  ) => value is Element & { type: string }
```

Comment:
I would suggest maybe using a Generic type for this:

```suggestion
  isElementType: <T extends Element>(
    value: any,
    discriminantVal: string,
    discriminantKey?: string
  ) => value is T
```

Reply:
implemented, and also im adding default Extend value to `Element & {type: string}`